### ISSUE TITLE
[Warn] Conflict of codegen methods

### ISSUE THREAD

**Issue by Devjiu** [intel-ai/hdk#569]
I am not sure how to resolve this clash of namings. 
```c++ 
ReductionCode ResultSetReductionJIT::codegen() const
```
vs
```c++
ReductionCode GpuReductionHelperJIT::codegen(const CompilationOptions& co) 
``` 
at `omniscidb/QueryEngine/ResultSetReductionJIT.h `

Second one method have single call, if it's what's intended to be done, let's remove virtual and rename `GpuReductionHelperJIT` method.

**Comment by alexbaden** [comment]
Can we add the unused `CompilationOptions` argument to the virtual method?

**Comment by Devjiu** [comment]
> Can we add the unused `CompilationOptions` argument to the virtual method?

Yes, this is one option, but I don't understand the `GpuReductionHelperJIT` class. As far as I understand it doesn't work as an inherited class at the moment. So if it's just composition and the `GpuReductionHelperJIT` needs to be used in a specific place, we can leave the `ResultSetReductionJIT` interface unchanged.
